### PR TITLE
updatecli: dynamic specs 

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

* Use `values.yml` to define the configuration to be used when running the `updatecli apply`, see https://www.updatecli.io/docs/commands/updatecli_apply/#updatecli-apply
* Use scm to fetch the specs in apm and apm-data to fetch all the files under the different folders.
* Use tar with the shell target to untar the file and update those files.

### Test

```bash
GITHUB_WORKSPACE=/tmp \
  PATH=/opt/homebrew/bin:$PATH \
  GIT_USER="Victor Martinez" \
  GIT_EMAIL="VictorMartinezRubio@gmail.com" \
  GITHUB_TOKEN=$(gh auth token) \
  updatecli apply --config .ci/updatecli/updatecli.d --values .ci/updatecli/values.yml --debug
```

or without pushing any changes

```bash
GITHUB_WORKSPACE=/tmp \
  PATH=/opt/homebrew/bin:$PATH \
  GIT_USER="Victor Martinez" \
  GIT_EMAIL="VictorMartinezRubio@gmail.com" \
  GITHUB_TOKEN=$(gh auth token) \
  updatecli apply --config .ci/updatecli/updatecli.d --values .ci/updatecli/values.yml --push=false --debug
```

additionally, you can change the org from `elastic` to your own GitHub avatar.

https://github.com/elastic/apm-agent-php/pull/1124 was created accordingly

### CODE REVIEW

File: .ci/updatecli/updatecli.d/update-json-specs.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      title: '[Automation] Update JSON specs'
+
+targets:
+  agent-json-specs:
+    name: APM agent json specs {{ source "sha" }}
+    scmid: default
+    sourceid: sha
+    kind: shell
+    spec:
+      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-specs.tgz && git --no-pager diff # required to ignore the sourceid being appended'
```
Comment (v1v) [inline .ci/updatecli/updatecli.d/update-json-specs.yml:81]:
```suggestion
    disablesourceinput: true
    kind: shell
    spec:
      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-specs.tgz && git --no-pager diff'
```

File: .ci/updatecli/updatecli.d/update-specs.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      title: '[Automation] Update JSON schema specs'
+
+targets:
+  agent-json-schema:
+    name: APM agent json server schema {{ source "sha" }}
+    scmid: default
+    sourceid: sha
+    kind: shell
+    spec:
+      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-schema.tgz && git --no-pager diff # required to ignore the sourceid being appended'
```
Comment (v1v) [inline .ci/updatecli/updatecli.d/update-specs.yml:83]:
```suggestion
    disablesourceinput: true
    kind: shell
    spec:
      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-schema.tgz && git --no-pager diff'
```

File: .ci/updatecli/updatecli.d/update-gherkin-specs.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      title: '[Automation] Update Gherkin specs'
+
+targets:
+  agent-gherkin-specs:
+    name: APM agent gherkin specs {{ source "sha" }}
+    scmid: default
+    sourceid: sha
+    kind: shell
+    spec:
+      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/gherkin-specs.tgz && git --no-pager diff # required to ignore the sourceid being appended'
```
Comment (v1v) [inline .ci/updatecli/updatecli.d/update-gherkin-specs.yml:81]:
```suggestion
    disablesourceinput: true
    kind: shell
    spec:
      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/gherkin-specs.tgz && git --no-pager diff'
```

File: .github/workflows/updatecli.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
           vaultUrl: ${{ secrets.VAULT_ADDR }}
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
-          pipeline: .ci/updatecli.d
+          pipeline: .ci/updatecli/updatecli.d
```
Comment (v1v) [inline .github/workflows/updatecli.yml:21]:
```suggestion
          pipeline: .ci/updatecli/updatecli.d
          values: .ci/updatecli/values.yml
```

**Review by reakaleek — APPROVED** [review]

**Comment by intuibase** [comment]
Hey @v1v 
What is the background to this change?
What does this tool give us?

**Comment by v1v** [comment]
> What is the background to this change?

It used to have defined the json-specs and specs from `elastic/apm` and `elastic/apm-data` in a static manner, so if for some reason those specs were renamed then they will be required to be changed in the `updatecli` pipeline. That's not convenience when there are 8 APM Agents.

If you search for `Simplify tests specs update from APM to APM agents` in this PR you can find the link to the private GitHub repository with the context.

> What does this tool give us?

That particular automation was already put in place in https://github.com/elastic/apm-agent-php/pull/889

Back in the days, we used to centralise all those changes but we came up with the proposal to let the consumers own the automation.

**Pull request merged**